app/components/profile.jsx

```diff
@@ -19,73 +23,60 @@ export class Profile extends Component {
     this.reset();
   }
 
-  componentWillUpdate (props) {
-    this.reset(props);
+  componentWillUpdate ({ params: { username } }) {
+    if (username !== this.props.params.username) {
+      this.reset();
```

Comment:
this is gonna be why we needed `nextProps` to be intact. we're actually going to be passing in these new props to the `reset` function. currently, the `reset` function doesnt accept any arguments, but we'll be changing that.

the reason why we need to do that is actually indicated from the previous line. what the `componentWillReceiveProps` (and `componentWillUpdate` for that matter) function is meant for is for you to get a sneak preview as to what the props _will_ be, not what it currently is. `this.props` is still the original props (including the original username and whatnot) and since `reset` uses `this.props`, when we call it, it literally does nothing. it just refetches the already-fetched user.

so. that means we need to give this sneak preview of the new props to the reset function and conditionally use that instead of `this.props`. if that doesnt make sense, let me know!!

final result:

```js
this.reset(nextProps);
```

you should also do this in `tracker.jsx` (line 30)